source/client_channel_handler.c

```diff
@@ -160,10 +136,37 @@ static int s_packet_handler_connack(
             "id=%p: invalid connect return code %d, disconnecting",
             (void *)connection,
             connack.connect_return_code);
-        /* If error code returned, disconnect */
+        /* If error code returned, disconnect, on_completed will be invoked from shutdown process */
         aws_channel_shutdown(connection->slot->channel, AWS_ERROR_MQTT_PROTOCOL_ERROR);
```

Comment:
This isn't a protocol error. We should really declare distinct error codes for each of the CONNACK non-zeroes.
AWS_ERROR_MQTT_CONNECTION_REFUSED_UNACCEPTABLE_PROTOCOL_VERSION???